spec/main_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require './lib/main'
+
+RSpec.describe Enumerable do
+  describe '#my_count' do
+    it 'calls a block on an object and returns the number of elements' do
+      arr = [1, 2, 3, 4, 5]
+      expect(arr.my_count).to eql(5)
```

Comment:
:thinking: What happens when a block is passed? Or an argument is passed?
Please, let's add edge cases for the test. This test is not yet effective as it passes even like this (Commenting out part of the code). :point_down: 
```ruby
def my_count(&block)
    # result = []
    return size unless block_given?

    # my_each do |elem|
    #   result << elem if block.call(elem) == true
    # end
    # result.size
  end
```